GliaWidgets/Sources/ViewModel/EngagementViewModel.swift

```diff
@@ -132,6 +132,7 @@ class EngagementViewModel: CommonEngagementModel {
                 }
                 self.engagementAction?(.showAlert(.operatorEndedEngagement(action: { [weak self] in
                     self?.endSession()
+                    self?.interactor.currentEngagement = nil
```

Comment:
The source of truth for `interactor.currentEngagement` is `coreSdkClient.getCurrentEngagement`. So `interactor` should update `currentEngagement` itself based on CoreSDK events, like engagement ended, media upgrades and so on.
Is it reasonable?

Reply:
I guess having something like `currentEngagementProducer` that will emit events on each change for `currentEngagement` in CoreSDK, might be even more useful and then we could get rid of `currentEngagement` property in `Interactor` at all.
@igorkravchenko @rasmustautsglia @ykyivskyi-gl what do you think?

Reply:
I will answer all your comments here since they are all linked.

Good points. If we can rely on core SDK for current engagement updates, then there is no need for any of this logic indeed. This seems like a good approach @EgorovEI

Reply:
I agree. We should keep updating currentEngagement from CoreSDK only. Let's see if you could use Combine instead of ReactiveSwift for emitting events.

Reply:
@EgorovEI do you mean introducing currentEngagementProducer on CoreSDK side?